client/components/LearnerViewDirective.js

```diff
@@ -553,9 +588,28 @@ tie.directive('learnerView', [function() {
             } else if (syntaxErrorIndex === null) {
               $scope.syntaxErrorString = '';
               $scope.syntaxErrorFound = false;
+
+              // Updating reinforcement bullets only if no syntax errors
+              reinforcement.clear();
+              for (var caseList in reinforcementObject.passedList) {
+                if (reinforcementObject.passedList[caseList]) {
```

Comment:
The reinforcement object should be fully prepared in a service and just passed to the controller -- there should be none of this logic in the directive's controller.

If necessary, you can handle the "don't show if there are syntax errors" bit with an if statement -- but that's about as much logic as should exist at this level of the code.